logfire/_internal/config.py

```diff
@@ -893,21 +893,33 @@ def add_span_processor(span_processor: SpanProcessor) -> None:
                 # try loading credentials (and thus token) from file if a token is not already available
                 # this takes the lowest priority, behind the token passed to `configure` and the environment variable
                 if self.token is None:
-                    credentials = LogfireCredentials.load_creds_file(self.data_dir)
-
-                    # if we still don't have a token, try initializing a new project and writing a new creds file
-                    # note, we only do this if `send_to_logfire` is explicitly `True`, not 'if-token-present'
-                    if self.send_to_logfire is True and credentials is None:
-                        credentials = LogfireCredentials.initialize_project(
-                            logfire_api_url=self.advanced.base_url,
-                            session=requests.Session(),
-                        )
-                        credentials.write_creds_file(self.data_dir)
-
-                    if credentials is not None:
-                        self.token = credentials.token
-                        self.advanced.base_url = self.advanced.base_url or credentials.logfire_api_url
-
+                    try:
+                        credentials = LogfireCredentials.load_creds_file(self.data_dir)
+
+                        # if we still don't have a token, try initializing a new project and writing a new creds file
+                        # note, we only do this if `send_to_logfire` is explicitly `True`, not 'if-token-present'
+                        if self.send_to_logfire is True and credentials is None:
+                            credentials = LogfireCredentials.initialize_project(
+                                logfire_api_url=self.advanced.base_url,
+                                session=requests.Session(),
+                            )
+                            credentials.write_creds_file(self.data_dir)
+
+                        if credentials is not None:
+                            self.token = credentials.token
+                            self.advanced.base_url = self.advanced.base_url or credentials.logfire_api_url
+                    except LogfireConfigError:
+                        if self.advanced.base_url is not None:
```

Comment:
I think we should only allow this if it is a grpc url?

Alternatively, perhaps we could split the logic, and have different code paths depending on whether the endpoint is HTTP/gRPC